### PR TITLE
#WP4NQf2S-ft-assign-driver-to-bus

### PR DESCRIPTION
**Pull request by Mugema123**
**What does this PR do?**
 PR is for assigning a driver to a bus
**Description of Task to be completed?**

- Operator should be able to assign and unassign a driver to a bus
` localhost:3020/api/v1/assignDriver/:id`
`localhost:3020/api/v1/unassignDriver/:id`
**How should this be manually tested?**

- By using postman (currently)
`login as Operator:'email:Josh123@gmail.com' and 'password:admin'`
`Driver email: use mugema8@gmail.com and bus Id use 2`
`Operator can assign or unassign a driver to a bus`
**What are the relevant pivotal tracker stories?**
#WP4NQf2S

**Screenshots (if appropriate)**

final1
final2

### CODE REVIEW

**Review by TuyizeeAnastase — CHANGES_REQUESTED** [review]
there are two files with the same code and functionalities(duplicated) `userController vs registController` and `busController vs bus.controller`

**Comment by TuyizeeAnastase** [comment]
i can't login using credentials given in description `login as admin:'email:josh@gmail.com' and 'password:admin'` 
but i used ```
{
    "email":"bruce@gmail.com",
    "password":"Driver"
}
``` and i think you can edit description.

File: src/database/migrations/20201020051502-create-users.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      role: { type: Sequelize.STRING, allowNull: false },
+      resetlink: { type: Sequelize.STRING(2000), },
+      createdAt: { allowNull: false, type: Sequelize.DATE },
+      updatedAt: { allowNull: false, type: Sequelize.DATE },
+    });
+  },
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.dropTable('Users');
+  },
+};
```
Comment (blaise82) [inline src/database/migrations/20201020051502-create-users.js]: why did you create a new user migration?

File: src/database/migrations/20201013145009-create-bus.js [hunk 2]
```diff
@@ -31,6 +31,9 @@ module.exports = {
       routeId: {
         type: Sequelize.INTEGER,
       },
+      userId: {
+        type: Sequelize.STRING
+      },
```
Comment (blaise82) [inline src/database/migrations/20201013145009-create-bus.js:36]: don't directly update the migration file

File: src/database/migrations/20201111141736-create-bus.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.dropTable('Buses');
+  }
+};
```
Comment (blaise82) [inline src/database/migrations/20201111141736-create-bus.js]: why do we have to create bus migration files?

**Review by blaise82 — CHANGES_REQUESTED** [review]

**Comment by Ally4** [comment]
You have conflicts
Assign bus does not work ("message": "provide valid email")
Unassign bus does not work ("message": "provide valid email")

**Comment by Ally4** [comment]
You have conflicts

**Review by Ally4 — APPROVED** [review]

**Pull request merged**